src/modules/QtQml/Timer.js

```diff
@@ -27,14 +27,21 @@ registerQmlType({
         }
     }
 
-    this.start = function() {
-        if (!this.running) {
-            this.running = true;
-            prevTrigger = (new Date).getTime();
+    /* This ensures that if the user toggles the "running" property manually,
+     * the timer will trigger. */
+    this.runningChanged.connect(function() {
+        if (self.running) {
+            prevTrigger = new Date().getTime();
             if (this.triggeredOnStart) {
```

Comment:
We need refactoring, code cleanup, and enforcing lint checks. That would make spotting such things a lot easier =).